plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Tag visitor that optimizes VIDEO tags:
+ * - Adds preload links for poster images if in a breakpoint group's LCP.
+ *
+ * @package image-prioritizer
+ *
+ * @since n.e.x.t
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Image Prioritizer: Image_Prioritizer_Video_Tag_Visitor class
+ *
+ * @since n.e.x.t
+ *
+ * @access private
+ */
+final class Image_Prioritizer_Video_Tag_Visitor extends Image_Prioritizer_Tag_Visitor {
+
+	/**
+	 * Visits a tag.
+	 *
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context.
+	 *
+	 * @return bool Whether the tag should be tracked in URL metrics.
+	 */
+	public function __invoke( OD_Tag_Visitor_Context $context ): bool {
+		$processor = $context->processor;
+		if ( 'VIDEO' !== $processor->get_tag() ) {
+			return false;
+		}
+
+		// Skip empty poster attributes and data: URLs.
+		$poster = trim( (string) $processor->get_attribute( 'poster' ) );
+		if ( '' === $poster || $this->is_data_url( $poster ) ) {
+			return false;
+		}
+
+		$xpath = $processor->get_xpath();
+
+		// TODO: If $context->url_metric_group_collection->get_element_max_intersection_ratio( $xpath ) is 0.0, then the video is not in any initial viewport and the VIDEO tag could get the preload=none attribute added.
```

Comment:
Thinking out loud for a future enhancement:

[`autoplay` takes precedence over `preload`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#preload). Since we here know that the video is not in the initial viewport, we could remove `autoplay` in favor of an `IntersectionObserver`-based implementation that only autoplays the video when actually visible.

Thoughts? Worth creating an issue for it?

Reply:
Oh, interesting idea. Yes, I think that warrants an issue. 

Related to #1035 which involves `IntersectionObserver` to lazy-load background images.

Reply:
Done: #1590